File: solutions/python/secret-handshake/3/secret_handshake.py
def commands(binary_str):
    actions = ["wink", "double blink", "close your eyes", "jump"]
    res = []
    
    for i in range(4):
        if binary_str[4 - i] == '1':
            res.append(actions[i])
            
    if binary_str[0] == '1':
        res.reverse()
        
    return res

File: solutions/ruby/secret-handshake/1/secret_handshake.rb
class SecretHandshake
  ACTIONS = ['wink', 'double blink', 'close your eyes', 'jump']

  def initialize(code)
    @code = code
  end

  def commands
    return [] if @code < 1
    result = []
    ACTIONS.each_with_index do |action, i|
      result << action if @code & (1 << i) != 0
    end
    @code & 16 != 0 ? result.reverse : result
  end
end
